instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/StartTimeExtractor.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.api.instrumenter;
+
+import java.time.Instant;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+/**
+ * Extractor of the start time of request processing. A {@link StartTimeExtractor} should always use
+ * the same timestamp source as the corresponding {@link EndTimeExtractor} - extracted timestamps
+ * must be comparable.
+ */
+@FunctionalInterface
+public interface StartTimeExtractor<REQUEST> {
+
+  /**
+   * Returns the default {@link StartTimeExtractor}, which delegates to the OpenTelemetry SDK
+   * internal clock.
+   */
+  static <REQUEST> StartTimeExtractor<REQUEST> getDefault() {
+    return request -> null;
+  }
+
+  /**
+   * Returns the timestamp marking the start of the request processing. If the returned timestamp is
+   * {@code null} the OpenTelemetry SDK will use its internal clock to determine the start time.
+   */
+  @Nullable
```

Comment:
Do you think there's a use case for a custom extractor to delegate to SDK conditionally? Otherwise maybe we don't need to make this default and can treat that as an implementation detail (we could even change the `null` check to `== getDefault()`. This is ok too though.

Reply:
> Do you think there's a use case for a custom extractor to delegate to SDK conditionally?

No, it doesn't make any sense for a custom extractor to delegate to SDK. I'll change that.
Thanks!

Reply:
I've removed the `getDefault()` methods from start/end extractors.